### PR TITLE
Rewrite maui tests

### LINKED ISSUE #3617: MAUI tests fail frequently and aren't easily diagnosable
These are an attempt to shim in the old internal tests (note that MAUI here is an old internal test framework). The old tests worked quite differently and don't fit in well with the current test infrastructure. We're getting [frequent intermittent failures](https://dev.azure.com/dnceng/public/_build/results?buildId=737706&view=ms.vss-test-web.build-test-results-tab&runId=22844842&resultId=100000&paneView=history) like so:

```
System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(scenarioName: \"Hotkey_Fires_OnClick\")

Assert.Equal() Failure\r\n          ↓ (pos 0)\r\nExpected: Pass\r\nActual:   Fail\r\n          ↑ (pos 0)

Stack trace
   at System.Windows.Forms.Maui.IntegrationTests.MauiTestHelper.ValidateScenarioPassed(String projectName, String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/MauiTestHelper.cs:line 90
   at System.Windows.Forms.Maui.IntegrationTests.WinformsMauiButtonTests.MauiButtonTest(String scenarioName) in /_/src/System.Windows.Forms/tests/IntegrationTests/System.Windows.Forms.Maui.IntegrationTests/WinformsMauiButtonTests.cs:line 35
``` 

This doesn't really help you much and it isn't obvious as to what you can do to move forward. In addition, the MAUI infrastructure sources aren't available so it is difficult to understand how the whole system works.

My recommendation (and current POR):

1. Disable the MAUI tests for now
2. Make them more xUnit-like and have any shim code in the repo
3. If we can make the tests plug in as a "normal" xUnit test then re-enable them for the CI, otherwise leave them for manual runs

cc: @AdamYoblick, @RussKie 

Failed with #3600

### PR DESCRIPTION
**Pull request by Danil-Andrianov**
Fixes #3617

## Proposed changes

- MAUI tests was removed
- Integration tests were added instead of them

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Manual
- CTI

 

## Test environment(s) 

- OS version: 10.0.19042.1348
- .NET 6.0.100-rc.2.21505.57

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6180)

### CODE REVIEW

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+namespace System.Windows.Forms.UI.IntegrationTests
+{
+    [ConfigureJoinableTaskFactory]
+    public class ComboBoxTests
+    {
+        private const int NumItems = 200;
+
+        private RandomUtil randomUtil;
+
+        private int numEvents;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:19]: [naming]

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+
+            return true;
+        }
+
+        [StaFact]
+        public void Add_Items_By_Index()
+        {
+            RunTest(comboBox =>
+            {
+            InitializeItems(comboBox);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:157]: [indentation]

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+        public void Verify_OnMeasureItem_receives_correct_arguments()
+        {
+            RunTestForDerivedComboBox(derivedComboBox =>
+            {
+                derivedComboBox.Items.AddRange(new object[]
+                {
+                "One",
+                "Two",
+                "Three"
+                });
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:312]: ditto

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/System.Windows.Forms.UI.IntegrationTests.csproj [hunk 4]
```diff
@@ -20,6 +20,12 @@
     <ProjectReference Include="..\System.Windows.Forms.IntegrationTests.Common\System.Windows.Forms.IntegrationTests.Common.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Reference Include="WinformsTestLib">
+      <HintPath>..\MauiTests\lib\WinformsTestLib.dll</HintPath>
+    </Reference>
+  </ItemGroup>
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/System.Windows.Forms.UI.IntegrationTests.csproj:27]: The test must require no external dependencies

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+            randomUtil = new();
+        }
+
+        private bool InitializeItems(ComboBox comboBox)
+        {
+            var szItems = CreateItems(NumItems);
+            return AddItems(comboBox, szItems, NumItems);
+        }
+
+        string[] CreateItems(int numItems)
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:32]: All members must have correct accessors applied.

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+                var index = randomUtil.GetRange(0, count - 1);
+                var item = (string)((index < count - 1) ? comboBox.Items[index + 1] : comboBox.Items[index]);
+
+                if (isByIndex)
+                    comboBox.Items.RemoveAt(index);
+                else
+                    comboBox.Items.Remove(comboBox.Items[index]);
+
+                if (comboBox.Items.Count != count - 1)
+                    return false;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:84]: Curlies are missing

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/System.Windows.Forms.UI.IntegrationTests.csproj (hunk 4 above)
Comment (Danil-Andrianov) [reply src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/System.Windows.Forms.UI.IntegrationTests.csproj:27]: But I really need this one, it provides RandomUtil and WFCTestLib. How is it possible to omit that dependency? Could you please share your opinion?
image

**Comment by RussKie** [comment]
With https://github.com/dotnet/winforms/pull/6091 merged, please port the remaining projects in a piecemeal way. Thank you.

**Pull request closed without merging**